pkg/stanza/operator/helper/transformer.go

```diff
@@ -106,10 +106,6 @@ func (t *TransformerOperator) HandleEntryError(ctx context.Context, entry *entry
 		}
 	}
 
-	if t.OnError == SendOnErrorQuiet || t.OnError == DropOnErrorQuiet {
-		return nil
-	}
```

Comment:
Why is this removed?

Reply:
The error returned by this method isn't soley used to log a message. Sometimes the error is read and based on that we decide if a log entry should be dropped or not.
A concrete example would be https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.112.0/pkg/stanza/operator/helper/parser.go#L111

I was fearful of any refactoring because I don't want to go against a possible design target.

Reply:
Won't think change result in a problem for transformers that are not parsers?

Reply:
I was not too fearful of removing this block, because it didn't exist before my previous PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35010
That being said, I see your point.
My rational is as such:
We can't replace an `error` with `nil` at this level as that would make it impossible for the caller to know if they should drop the entry or send it.
The caller has visibility on the config like `OnError`, so I ended moving the responsibility to the caller.

Let me know what you think.